File: modules/swagger-core/src/main/scala/com/wordnik/swagger/config/FilterFactory.scala
package com.wordnik.swagger.config

import com.wordnik.swagger.core.filter._
import com.wordnik.swagger.model._

object FilterFactory {
  var filter: SwaggerSpecFilter = new DefaultSpecFilter

  def setFilter(filter: SwaggerSpecFilter) = {
    FilterFactory.filter = filter
  }

  def getFilter() = {
    filter
  }
}

class DefaultSpecFilter extends SwaggerSpecFilter {
  def isOperationAllowed(operation: Operation, api: ApiDescription, params: java.util.Map[String, java.util.List[String]], cookies: java.util.Map[String, String], headers: java.util.Map[String, java.util.List[String]]): Boolean = true
  def isParamAllowed(parameter: Parameter, operation: Operation, api: ApiDescription, params: java.util.Map[String, java.util.List[String]], cookies: java.util.Map[String, String], headers: java.util.Map[String, java.util.List[String]]): Boolean = true
}

File: modules/swagger-core/src/main/java/com/wordnik/swagger/config/FilterFactory.java
package com.wordnik.swagger.config;

import com.wordnik.swagger.core.filter.*;

public class FilterFactory {
  protected static SwaggerSpecFilter FILTER = null;

  public static void setFilter(SwaggerSpecFilter filter) {
    FILTER = filter;
  }
  public static SwaggerSpecFilter getFilter() {
    return FILTER;
  }
}